fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneRecordCursor.java

```diff
@@ -288,16 +214,52 @@ private synchronized IndexReader getIndexReader() throws IOException {
         return FDBDirectoryManager.getManager(state).getIndexReader(groupingKey);
     }
 
-    private void performScan() throws IOException {
+    private void maybePerformScan() throws IOException {
+        if (lookupResults != null && currentPosition + skip < pageSize) {
```

Comment:
`currentPosition` is an absolute position, not an in-current-page position. Therefore I do not believe it is comparable with `pageSize` after the first page.

Reply:
yeah this if condition can be completely removed now.